js/script.js

```diff
@@ -1,11 +1,28 @@
 "use strict";
 
-// Global object to store currency rates
-const currencyRates = {
-  timestamp: Date.now(),
-  base: "EUR",
-  date: new Date().toISOString().split("T")[0],
-  rates: {},
+// Global array to store currency rates
+const currencyRates = [];
+
+// Function to find a rate
+const findRate = (base, target) => {
```

Comment:
This function looks like a weaker version of `searchRate` (line 147). I recommend removing it and using `searchRate` instead.